### DISCUSSION TITLE
Initial slight delay, then none?

### DISCUSSION THREAD

**Discussion by IRH1980** [jason0x43/hacs-hubitat#217] [General]
I was wondering if there was a reason for the behaviour i'm seeing, controlling zigbee devices which are attached to Hubitat. When i first toggle a light having left it alone for a while, there is a slight delay. If i toggle the next light (or the same one again) soon after it will then operate without this delay. Is there a concept of something along the chain 'going to sleep' after a period of inactivity?

**Comment by jason0x43** [comment]
I'm not aware of anything on the Home Assistant side that would cause a delay like that (but that doesn't mean there isn't something there).

If you can experiment a bit, it might help narrow down the issue:
* Do you see similar delays if you try to control one of the zigbee devices directly through Hubitat after not using it for a while?
* Do the delays happen for any zigbee device, or only certain ones?
* Do the delays happen for all devices, or only zigbee devices?

**Reply by IRH1980** [reply]
I will investigate and report back. Might take a little while as i dont know how long i have to leave it for the delay to kick in again.

**Reply by IRH1980** [reply]
I have had a play with this. At the moment i have limited devices on the hubitat (see below why), only 1 x Hue Vintage and 1 x Ikea Tradfri outlet. They are in the same room as the Hubitat hub.

Do you see similar delays if you try to control one of the zigbee devices directly through Hubitat after not using it for a while?
No - they seem to be ok when worked from hubitat, whichever one i switch first.

Do the delays happen for any zigbee device, or only certain ones?
Acknowledging that two is a limited sample, they are both affected.

Do the delays happen for all devices, or only zigbee devices?
I only have these two Zigbee devices on the hubitat, i have no Z Wave devices.

By way of explanation, i used hubitat for a long time. I then made the switch to HA with skyconnect, but have seen some unreliability using ZHA and skyconnect, hence i though to trial the hubitat hub as a zigbee handler via HA. Most of the devices are still on HA, i just switched two to hubitat for testing and hit this delay issue.

If i can do anything else, logs etc, just let me know where to look.
